### PR TITLE
Checks consistency of cmake and submodule repo

### PR DESCRIPTION
**Pull request by bhourahine**
Changes script to optionally check for 
* Mismatch between CMakeLists.txt and .gitmodules urls
* Whether the git urls have consistent .git extension (i.e. headless or not)
* Update the 2 urls that point to the wrong places

### CODE REVIEW

File: utils/test/check_submodule_commits [hunk 1]
```diff
@@ -64,8 +74,10 @@ def main():
 def _parse_arguments():
     """Returns command line arguments"""
     parser = argparse.ArgumentParser(description=_DESCRIPTION)
-    msg = "Whether the commits should be updated in the CMakeLists.txt files"
-    parser.add_argument("-u", "--update", action='store_true', help=msg)
+    msgu = "Whether the commits should be updated in the CMakeLists.txt files"
+    parser.add_argument("-u", "--update", action='store_true', help=msgu)
+    msgc = "Check consistency between CMakeLists.txt and submodules"
```
Comment (aradi) [inline utils/test/check_submodule_commits:79]: Any reason for not just reusing the `msg` variable? (Once `parser.add_argument()` had been called, the content of `msg` had been copied to an internal variable of the argument parser.)

File: utils/test/check_submodule_commits [hunk 2]
```diff
@@ -33,23 +34,32 @@ def main():
...
+        gitconfig.read('.git/config')
     allsync = True
     with open("CMakeLists.txt", "r") as fp:
         origcontent = fp.read()
     content = origcontent
     for match in _PAT_SUBMOD_COMMIT.finditer(output):
         commit, submodule = match.groups()
+        if args.check:
+            locale = ('submodule "external/%s/origin"' % submodule)
+            active=gitconfig.get(locale, 'active')
```
Comment (aradi) [inline utils/test/check_submodule_commits:48]:
```suggestion
            active = gitconfig.get(locale, 'active')
```

File: utils/test/check_submodule_commits [hunk 3]
```diff
@@ -33,23 +34,32 @@ def main():
...
+            if (active == 'true'):
```
Comment (aradi) [inline utils/test/check_submodule_commits:49]:
```suggestion
            if active == 'true':
```

**Review by aradi — APPROVED** [review]
Thx, green light from my side, I have only minor stylistic suggestions.

File: utils/test/check_submodule_commits (hunk 1 above)
Comment (bhourahine) [reply utils/test/check_submodule_commits]: msg now used for both

**Pull request merged**